File: src/main/python/net/coderextreme/data/ball.py
print('<!--')
import x3d
print('-->')
X3D0 = x3d.X3D()
X3D0.profile = "Immersive"
X3D0.version = "4.0"
head1 = x3d.head()
component2 = x3d.component()
component2.name = "EnvironmentalEffects"
component2.level = 3

head1.children.append(component2)
component3 = x3d.component()
component3.name = "Shaders"
component3.level = 1

head1.children.append(component3)
component4 = x3d.component()
component4.name = "CubeMapTexturing"
component4.level = 1

head1.children.append(component4)
component5 = x3d.component()
component5.name = "Texturing"
component5.level = 1

head1.children.append(component5)
component6 = x3d.component()
component6.name = "Rendering"
component6.level = 1

head1.children.append(component6)
component7 = x3d.component()
component7.name = "Grouping"
component7.level = 3

head1.children.append(component7)
component8 = x3d.component()
component8.name = "Core"
component8.level = 1

head1.children.append(component8)
meta9 = x3d.meta()
meta9.name = "title"
meta9.content = "ball.x3d"

head1.children.append(meta9)
meta10 = x3d.meta()
meta10.name = "creator"
meta10.content = "John Carlson"

head1.children.append(meta10)
meta11 = x3d.meta()
meta11.name = "generator"
meta11.content = "manual"

head1.children.append(meta11)
meta12 = x3d.meta()
meta12.name = "identifier"
meta12.content = "https://coderextreme.net/X3DJSONLD/src/main/data/ball.x3d"

head1.children.append(meta12)
meta13 = x3d.meta()
meta13.name = "description"
meta13.content = "a prismatic sphere"

head1.children.append(meta13)

X3D0.head = head1
Scene14 = x3d.Scene()
WorldInfo15 = x3d.WorldInfo()
WorldInfo15.title = "ball.x3d"

Scene14.children.append(WorldInfo15)
NavigationInfo16 = x3d.NavigationInfo()
NavigationInfo16.type = ["ANY","EXAMINE","FLY","LOOKAT"]

Scene14.children.append(NavigationInfo16)
Viewpoint17 = x3d.Viewpoint()
Viewpoint17.description = "Tour Views"
Viewpoint17.position = [0,0,12]

Scene14.children.append(Viewpoint17)
Background18 = x3d.Background()
Background18.backUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_back.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_back.png"]
Background18.bottomUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_bottom.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_bottom.png"]
Background18.frontUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_front.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_front.png"]
Background18.leftUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_left.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_left.png"]
Background18.rightUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_right.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_right.png"]
Background18.topUrl = ["../resources/images/all_probes/stpeters_cross/stpeters_top.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_top.png"]

Scene14.children.append(Background18)
Transform19 = x3d.Transform()
Shape20 = x3d.Shape()
Sphere21 = x3d.Sphere()

Shape20.geometry = Sphere21
Appearance22 = x3d.Appearance()
Material23 = x3d.Material()
Material23.diffuseColor = [0.7,0.7,0.7]
Material23.specularColor = [0.5,0.5,0.5]

Appearance22.material = Material23
ComposedCubeMapTexture24 = x3d.ComposedCubeMapTexture(DEF="texture")
ImageTexture25 = x3d.ImageTexture()
ImageTexture25.url = ["../resources/images/all_probes/stpeters_cross/stpeters_back.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_back.png"]

ComposedCubeMapTexture24.backTexture = ImageTexture25
ImageTexture26 = x3d.ImageTexture()
ImageTexture26.url = ["../resources/images/all_probes/stpeters_cross/stpeters_bottom.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_bottom.png"]

ComposedCubeMapTexture24.bottomTexture = ImageTexture26
ImageTexture27 = x3d.ImageTexture()
ImageTexture27.url = ["../resources/images/all_probes/stpeters_cross/stpeters_front.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_front.png"]

ComposedCubeMapTexture24.frontTexture = ImageTexture27
ImageTexture28 = x3d.ImageTexture()
ImageTexture28.url = ["../resources/images/all_probes/stpeters_cross/stpeters_left.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_left.png"]

ComposedCubeMapTexture24.leftTexture = ImageTexture28
ImageTexture29 = x3d.ImageTexture()
ImageTexture29.url = ["../resources/images/all_probes/stpeters_cross/stpeters_right.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_right.png"]

ComposedCubeMapTexture24.rightTexture = ImageTexture29
ImageTexture30 = x3d.ImageTexture()
ImageTexture30.url = ["../resources/images/all_probes/stpeters_cross/stpeters_top.png","https://coderextreme.net/X3DJSONLD/src/main/resources/images/all_probes/stpeters_cross/stpeters_top.png"]

ComposedCubeMapTexture24.topTexture = ImageTexture30

Appearance22.texture = ComposedCubeMapTexture24
Appearance22.children.append(x3d.Comment("""<ProgramShader DEF='ProgramShader' containerField='shaders' language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<ShaderProgram url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/freewrl.vs\"' containerField='programs'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'/>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'/>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'/>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='initializeOnly' type='SFFloat' value='2'/>"""))
Appearance22.children.append(x3d.Comment("""</ShaderProgram>"""))
Appearance22.children.append(x3d.Comment("""<ShaderProgram url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/freewrl.fs\"' containerField='programs' type='FRAGMENT'/>"""))
Appearance22.children.append(x3d.Comment("""</ProgramShader>"""))
Appearance22.children.append(x3d.Comment("""<ComposedShader language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='inputOutput' type='SFVec3f' value='0.98 1 1.033'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='cube' type='SFNode' accessType=\"inputOutput\">"""))
Appearance22.children.append(x3d.Comment("""<ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture>"""))
Appearance22.children.append(x3d.Comment("""</field>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='inputOutput' type='SFFloat' value='2'></field>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart url='\"../shaders/castle.vs\" \"https://coderextreme.net/X3DJSONLD/src/main/shaders/castle.vs\"'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart DEF='commonfs' url='\"../shaders/common.fs\" \"https://coderextreme.net/X3DJSONLD/src/main/shaders/common.fs\"' type='FRAGMENT'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""</ComposedShader>"""))
Appearance22.children.append(x3d.Comment("""<ComposedShader language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='fw_Texture_unit0' type='SFNode' accessType=\"initializeOnly\">"""))
Appearance22.children.append(x3d.Comment("""<ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture>"""))
Appearance22.children.append(x3d.Comment("""</field>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='initializeOnly' type='SFFloat' value='2'></field>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/contact.vs\"'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart USE='commonfs'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""</ComposedShader>"""))
Appearance22.children.append(x3d.Comment("""<ComposedShader language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='inputOutput' type='SFVec3f' value='0.98 1 1.033'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='cube' type='SFNode' accessType=\"inputOutput\">"""))
Appearance22.children.append(x3d.Comment("""<ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture>"""))
Appearance22.children.append(x3d.Comment("""</field>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='inputOutput' type='SFFloat' value='2'></field>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/octaga.vs\"'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart USE='commonfs'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""</ComposedShader>"""))
Appearance22.children.append(x3d.Comment("""<ComposedShader language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='cube' accessType='initializeOnly' type='SFInt32' value='0'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='initializeOnly' type='SFFloat' value='2'></field>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/instant.vs\"'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart USE='commonfs'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""</ComposedShader>"""))
Appearance22.children.append(x3d.Comment("""<ComposedShader language='GLSL'>"""))
Appearance22.children.append(x3d.Comment("""<field name='chromaticDispertion' accessType='inputOutput' type='SFVec3f' value='0.98 1 1.033'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='cube' type='SFNode' accessType=\"inputOutput\">"""))
Appearance22.children.append(x3d.Comment("""<ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture>"""))
Appearance22.children.append(x3d.Comment("""</field>"""))
Appearance22.children.append(x3d.Comment("""<field name='bias' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='scale' accessType='inputOutput' type='SFFloat' value='0.5'></field>"""))
Appearance22.children.append(x3d.Comment("""<field name='power' accessType='inputOutput' type='SFFloat' value='2'></field>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart url='\"https://coderextreme.net/X3DJSONLD/src/main/shaders/x3dom.vs\"'></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""<ShaderPart USE=\"commonfs\"></ShaderPart>"""))
Appearance22.children.append(x3d.Comment("""</ComposedShader>"""))
ComposedShader31 = x3d.ComposedShader()
ComposedShader31.language = "GLSL"
field32 = x3d.field()
field32.name = "chromaticDispertion"
field32.accessType = "initializeOnly"
field32.type = "SFVec3f"
field32.value = [0.98,1,1.033]

ComposedShader31.field.append(field32)
field33 = x3d.field()
field33.name = "cube"
field33.type = "SFNode"
field33.accessType = "initializeOnly"
ComposedCubeMapTexture34 = x3d.ComposedCubeMapTexture(USE="texture")

field33.children.append(ComposedCubeMapTexture34)

ComposedShader31.field.append(field33)
field35 = x3d.field()
field35.name = "bias"
field35.accessType = "initializeOnly"
field35.type = "SFFloat"
field35.value = 0.5

ComposedShader31.field.append(field35)
field36 = x3d.field()
field36.name = "scale"
field36.accessType = "initializeOnly"
field36.type = "SFFloat"
field36.value = 0.5

ComposedShader31.field.append(field36)
field37 = x3d.field()
field37.name = "power"
field37.accessType = "initializeOnly"
field37.type = "SFFloat"
field37.value = 2

ComposedShader31.field.append(field37)
ShaderPart38 = x3d.ShaderPart()
ShaderPart38.url = ["https://coderextreme.net/X3DJSONLD/src/main/shaders/x_ite.vs"]
ShaderPart38.type = "VERTEX"

ComposedShader31.parts.append(ShaderPart38)
ShaderPart39 = x3d.ShaderPart(DEF="commonfs")
ShaderPart39.url = ["https://coderextreme.net/X3DJSONLD/src/main/shaders/commonnew.fs"]
ShaderPart39.type = "FRAGMENT"

ComposedShader31.parts.append(ShaderPart39)
ComposedShader31.children.append(x3d.Comment("""<ShaderPart USE=\"commonfs\"></ShaderPart>"""))

Appearance22.shaders.append(ComposedShader31)

Shape20.appearance = Appearance22

Transform19.children.append(Shape20)

Scene14.children.append(Transform19)

X3D0.Scene = Scene14
f = open("../data/ball.new.python.x3d", mode="w", encoding="utf-8")
f.write(X3D0.XML())
f.close()
f = open("../data/ball.new.python.x3dv", mode="w", encoding="utf-8")
f.write(X3D0.VRML())
f.close()
f = open("../data/ball.new.python.x3dj", mode="w", encoding="utf-8")
f.write(X3D0.JSON())
f.close()

File: src/main/cplusplus/net/coderextreme/data/ball.cpp
/* delete next 5 lines for Visual Studio */
#define FALSE false
#define TRUE true
#define WINAPI
#define AFX_EXT_CLASS
#define EXPORT32
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/CylinderExample/pch.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/CylinderExample/framework.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/XML_PARSER.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/X3DLib.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/Abstracts.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/Concretes.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/glMath.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter02/CylinderExample/include/glut.h"
int main(int argc, char ** argv) {
X3D* X3D0 = new X3D();
X3D0->setProfile("Immersive");
X3D0->setVersion("4.0");
head* head1 = new head();
component* component2 = new component();
component2->setName("Scripting");
component2->setLevel(1);
head1->addComponent(component2);

component* component3 = new component();
component3->setName("EnvironmentalEffects");
component3->setLevel(3);
head1->addComponent(component3);

component* component4 = new component();
component4->setName("Shaders");
component4->setLevel(1);
head1->addComponent(component4);

component* component5 = new component();
component5->setName("CubeMapTexturing");
component5->setLevel(1);
head1->addComponent(component5);

component* component6 = new component();
component6->setName("Texturing");
component6->setLevel(1);
head1->addComponent(component6);

component* component7 = new component();
component7->setName("Rendering");
component7->setLevel(1);
head1->addComponent(component7);

component* component8 = new component();
component8->setName("Grouping");
component8->setLevel(3);
head1->addComponent(component8);

component* component9 = new component();
component9->setName("Core");
component9->setLevel(1);
head1->addComponent(component9);

meta* meta10 = new meta();
meta10->setName("title");
meta10->setContent("ball.x3d");
head1->addMeta(meta10);

meta* meta11 = new meta();
meta11->setName("creator");
meta11->setContent("John Carlson");
head1->addMeta(meta11);

meta* meta12 = new meta();
meta12->setName("generator");
meta12->setContent("manual");
head1->addMeta(meta12);

meta* meta13 = new meta();
meta13->setName("identifier");
meta13->setContent("https://coderextreme.net/X3DJSONLD/src/main/data/ball.x3d");
head1->addMeta(meta13);

meta* meta14 = new meta();
meta14->setName("description");
meta14->setContent("a prismatic sphere");
head1->addMeta(meta14);

X3D0->setHead(head1);

Scene* Scene15 = new Scene();
WorldInfo* WorldInfo16 = new WorldInfo();
WorldInfo16->setTitle("ball.x3d");
Scene15->addChild(WorldInfo16);

NavigationInfo* NavigationInfo17 = new NavigationInfo();
NavigationInfo17->setType(new String[4]{"ANY","EXAMINE","FLY","LOOKAT"}, 4);
Scene15->addChild(NavigationInfo17);

Viewpoint* Viewpoint18 = new Viewpoint();
Viewpoint18->setDescription("Tour Views");
Scene15->addChild(Viewpoint18);

Background* Background19 = new Background();
Background19->setBackUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_back.png"}, 1);
Background19->setBottomUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_bottom.png"}, 1);
Background19->setFrontUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_front.png"}, 1);
Background19->setLeftUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_left.png"}, 1);
Background19->setRightUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_right.png"}, 1);
Background19->setTopUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_top.png"}, 1);
Scene15->addChild(Background19);

Transform* Transform20 = new Transform();
Shape* Shape21 = new Shape();
Sphere* Sphere22 = new Sphere();
Shape21->setGeometry(Sphere22);

Appearance* Appearance23 = new Appearance();
Material* Material24 = new Material();
Material24->setDiffuseColor(new float[3]{0.7,0.7,0.7});
Material24->setSpecularColor(new float[3]{0.5,0.5,0.5});
Appearance23->setMaterial(Material24);

ComposedCubeMapTexture* ComposedCubeMapTexture25 = new ComposedCubeMapTexture();
ComposedCubeMapTexture25->setDEF("texture");
ImageTexture* ImageTexture26 = new ImageTexture();
ImageTexture26->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_back.png"}, 1);
ComposedCubeMapTexture25->setBackTexture(ImageTexture26);

ImageTexture* ImageTexture27 = new ImageTexture();
ImageTexture27->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_bottom.png"}, 1);
ComposedCubeMapTexture25->setBottomTexture(ImageTexture27);

ImageTexture* ImageTexture28 = new ImageTexture();
ImageTexture28->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_front.png"}, 1);
ComposedCubeMapTexture25->setFrontTexture(ImageTexture28);

ImageTexture* ImageTexture29 = new ImageTexture();
ImageTexture29->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_left.png"}, 1);
ComposedCubeMapTexture25->setLeftTexture(ImageTexture29);

ImageTexture* ImageTexture30 = new ImageTexture();
ImageTexture30->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_right.png"}, 1);
ComposedCubeMapTexture25->setRightTexture(ImageTexture30);

ImageTexture* ImageTexture31 = new ImageTexture();
ImageTexture31->setUrl(new String[1]{"../resources/images/all_probes/stpeters_cross/stpeters_top.png"}, 1);
ComposedCubeMapTexture25->setTopTexture(ImageTexture31);

Appearance23->setTexture(ComposedCubeMapTexture25);

//<ProgramShader DEF='ProgramShader' containerField='shaders' language='GLSL'> <ShaderProgram url='\"../shaders/freewrl.vs\"' containerField='programs' type='VERTEX'> <field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'/> <field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'/> <field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'/> <field name='power' accessType='initializeOnly' type='SFFloat' value='2'/> </ShaderProgram> <ShaderProgram url='\"../shaders/freewrl.fs\"' containerField='programs' type='FRAGMENT'/> </ProgramShader>
//<ComposedShader language='GLSL'> <field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'></field> <field name='fw_Texture_unit0' type='SFNode' accessType=\"initializeOnly\"> <ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture> </field> <field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'></field> <field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'></field> <field name='power' accessType='initializeOnly' type='SFFloat' value='2'></field> <ShaderPart url='\"../shaders/contact.vs\"' type=\"VERTEX\" containerField='parts'></ShaderPart> <ShaderPart url='\"../shaders/common.fs\"' containerField='parts' type='FRAGMENT'></ShaderPart> </ComposedShader>
//<ComposedShader language='GLSL'> <field name='chromaticDispertion' accessType='inputOutput' type='SFVec3f' value='0.98 1 1.033'></field> <field name='cube' type='SFNode' accessType=\"inputOutput\"> <ComposedCubeMapTexture USE=\"texture\"></ComposedCubeMapTexture> </field> <field name='bias' accessType='inputOutput' type='SFFloat' value='0.5'></field> <field name='scale' accessType='inputOutput' type='SFFloat' value='0.5'></field> <field name='power' accessType='inputOutput' type='SFFloat' value='2'></field> <ShaderPart url='\"../shaders/octaga.vs\"' type=\"VERTEX\" containerField='parts'></ShaderPart> <ShaderPart url='\"../shaders/common.fs\"' containerField='parts' type='FRAGMENT'></ShaderPart> </ComposedShader>
//<ComposedShader language='GLSL'> <field name='chromaticDispertion' accessType='initializeOnly' type='SFVec3f' value='0.98 1 1.033'></field> <field name='cube' accessType='initializeOnly' type='SFInt32' value='0'></field> <field name='bias' accessType='initializeOnly' type='SFFloat' value='0.5'></field> <field name='scale' accessType='initializeOnly' type='SFFloat' value='0.5'></field> <field name='power' accessType='initializeOnly' type='SFFloat' value='2'></field> <ShaderPart url='\"../shaders/instant.vs\"' type=\"VERTEX\" containerField='parts'></ShaderPart> <ShaderPart url='\"../shaders/common.fs\"' containerField='parts' type='FRAGMENT'></ShaderPart> </ComposedShader>
//
ComposedShader* ComposedShader32 = new ComposedShader();
ComposedShader32->setLanguage("GLSL");
field* field33 = new field();
field33->setName("chromaticDispertion");
field33->setAccessType("inputOutput");
field33->setType("SFVec3f");
field33->setValue("0.98 1 1.033");
ComposedShader32->addField(field33);

field* field34 = new field();
field34->setName("cube");
field34->setType("SFNode");
field34->setAccessType("inputOutput");
ComposedCubeMapTexture* ComposedCubeMapTexture35 = new ComposedCubeMapTexture();
ComposedCubeMapTexture35->setUSE("texture");
field34->addChild(ComposedCubeMapTexture35);

ComposedShader32->addField(field34);

field* field36 = new field();
field36->setName("bias");
field36->setAccessType("inputOutput");
field36->setType("SFFloat");
field36->setValue("0.5");
ComposedShader32->addField(field36);

field* field37 = new field();
field37->setName("scale");
field37->setAccessType("inputOutput");
field37->setType("SFFloat");
field37->setValue("0.5");
ComposedShader32->addField(field37);

field* field38 = new field();
field38->setName("power");
field38->setAccessType("inputOutput");
field38->setType("SFFloat");
field38->setValue("2");
ComposedShader32->addField(field38);

ShaderPart* ShaderPart39 = new ShaderPart();
ShaderPart39->setUrl(new String[1]{"../shaders/x3dom.vs"}, 1);
ShaderPart39->setType("VERTEX");
ComposedShader32->addParts(ShaderPart39);

ShaderPart* ShaderPart40 = new ShaderPart();
ShaderPart40->setDEF("common");
ShaderPart40->setUrl(new String[1]{"../shaders/common.fs"}, 1);
ShaderPart40->setType("FRAGMENT");
ComposedShader32->addParts(ShaderPart40);

Appearance23->addShaders(ComposedShader32);

ComposedShader* ComposedShader41 = new ComposedShader();
ComposedShader41->setLanguage("GLSL");
field* field42 = new field();
field42->setName("chromaticDispertion");
field42->setAccessType("initializeOnly");
field42->setType("SFVec3f");
field42->setValue("0.98 1 1.033");
ComposedShader41->addField(field42);

field* field43 = new field();
field43->setName("cube");
field43->setType("SFNode");
field43->setAccessType("initializeOnly");
ComposedCubeMapTexture* ComposedCubeMapTexture44 = new ComposedCubeMapTexture();
ComposedCubeMapTexture44->setUSE("texture");
field43->addChild(ComposedCubeMapTexture44);

ComposedShader41->addField(field43);

field* field45 = new field();
field45->setName("bias");
field45->setAccessType("initializeOnly");
field45->setType("SFFloat");
field45->setValue("0.5");
ComposedShader41->addField(field45);

field* field46 = new field();
field46->setName("scale");
field46->setAccessType("initializeOnly");
field46->setType("SFFloat");
field46->setValue("0.5");
ComposedShader41->addField(field46);

field* field47 = new field();
field47->setName("power");
field47->setAccessType("initializeOnly");
field47->setType("SFFloat");
field47->setValue("2");
ComposedShader41->addField(field47);

ShaderPart* ShaderPart48 = new ShaderPart();
ShaderPart48->setUrl(new String[1]{"../shaders/x_ite.vs"}, 1);
ShaderPart48->setType("VERTEX");
ComposedShader41->addParts(ShaderPart48);

ShaderPart* ShaderPart49 = new ShaderPart();
ShaderPart49->setUrl(new String[1]{"../shaders/x_itebubbles.fs"}, 1);
ShaderPart49->setType("FRAGMENT");
ComposedShader41->addParts(ShaderPart49);

Appearance23->addShaders(ComposedShader41);

Shape21->setAppearance(Appearance23);

Transform20->addChild(Shape21);

Scene15->addChild(Transform20);

X3D0->setScene(Scene15);

X3D0->toXMLString();
}
